src/main/java/org/folio/event/handler/HoldingCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,99 @@
+package org.folio.event.handler;
+
+import static org.folio.event.InventoryEventType.INVENTORY_HOLDING_CREATE;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.Logger;
+import org.folio.event.dto.InventoryFields;
+import org.folio.event.dto.ResourceEvent;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.rest.jaxrs.model.Location;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.persist.DBClient;
+import org.folio.services.lines.PoLinesService;
+import org.folio.services.piece.PieceService;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class HoldingCreateAsyncRecordHandler extends InventoryCreateAsyncRecordHandler {
+
+  @Autowired
+  private PieceService pieceService;
+
+  @Autowired
+  private PoLinesService poLinesService;
+
+  public HoldingCreateAsyncRecordHandler(Vertx vertx, Context context) {
+    super(INVENTORY_HOLDING_CREATE, vertx, context);
+    SpringContextUtil.autowireDependencies(this, context);
+  }
+
+  @Override
+  protected Future<Void> processInventoryCreationEvent(ResourceEvent resourceEvent, String tenantId) {
+    var holdingObject = JsonObject.mapFrom(resourceEvent.getNewValue());
+    var holdingId = holdingObject.getString(InventoryFields.ID.getValue());
+    var dbClient = new DBClient(getVertx(), tenantId);
+    var tenantIdUpdates = List.of(
+      processPoLinesUpdate(holdingId, tenantId, dbClient),
+      processPiecesUpdate(holdingId, tenantId, dbClient)
+    );
+    return GenericCompositeFuture.all(tenantIdUpdates).mapEmpty();
+  }
+
+  private Future<Void> processPoLinesUpdate(String holdingId, String tenantId, DBClient dbClient) {
```

Comment:
we need to store info about each po line update in mod-audit, so here we should use producer to push messages to kafka. You can see how to use it by searching auditOutboxService.saveOrderLinesOutboxLogs

Reply:
So propose to open 1 transaction and use that connection for po line updates, piece updates and saving po line details into audit log

Reply:
Should we not do the same for Pieces also?

Reply:
It is good question, basically in mod-audit we have an endpoint only for finding pieces with changed statuses. But in mod-orders-storage we always send events for app pieces operations. I think we should be consistent and do the same here for pieces as well.